File: nim/bubble_sort.nim
## Bubble Sort example in Nim Programming Language

## bubble_sort takes an unsorted sequence of integers sort the sequence
proc bubble_sort(input: var seq[int]) =
  for i in countup(1, input.len - 1):
    for j in countdown(input.len - 1, i):
      if input[j - 1] > input[j]:
        # swap the two numbers
        var temp = input[j]
        input[j] = input[j - 1]
        input[j - 1] = temp


var input: seq[int] = @[5, 3, 6, 2, 3, 4, 5]

echo input

bubble_sort(input)
echo input

File: java/bubble_sort.java
class bubble_sort() {

    public static int[] sort(int[] array) {
        int arrayEnd = array.length - 1;

        for(int i=0; i<arrayEnd; i++) {
            for(int j=0; j < arrayEnd; j++) {
                if(array[j] > array[j+1]) {
                    int temp = array[j+1];
                    array[j+1] = array[j];
                    array[j] = temp;
                }
            }
        }
        return array;
    }

}